plugin/trino-postgresql/src/main/java/io/trino/plugin/postgresql/PostgreSqlClient.java

```diff
@@ -307,6 +313,30 @@ public PostgreSqlClient(
                         .build());
     }
 
+    @Override
+    public PreparedStatement buildSql(ConnectorSession session, Connection connection, JdbcSplit split, JdbcTableHandle table, List<JdbcColumnHandle> columns)
+            throws SQLException
+    {
+        Map<String, String> supposedColumnExpressions = new HashMap<>();
+        for (JdbcColumnHandle column : columns) {
+            JdbcTypeHandle jdbcTypeHandle = column.getJdbcTypeHandle();
+            if (jdbcTypeHandle.getJdbcTypeName().isPresent() && jdbcTypeHandle.getJdbcTypeName().get().equals("geometry")) {
+                String columnName = column.getColumnName();
+                log.debug("Find geometry type, changing '%s' to '%s'", columnName, "ST_AsBinary(\"" + columnName + "\")");
+                supposedColumnExpressions.put(columnName, "ST_AsBinary(\"" + columnName + "\")");
```

Comment:
cc: @ebyhr since you recently were looking into ways to apply functions to columns on the read path. Do you know of a better alternative?